### PR TITLE
code snippets to use fh.auth in the app preview

### PR DESCRIPTION
**Pull request by rachael-oregan**
# Motivation 
Studio Preview/Local Development for Web Apps does not handle $fh.auth calls correctly

A new auth end was added to handle $fh.auth requests correctly in fh-mbaas-api.

Customer needs code snippets to show how the new endpoint should be used.

# Result
Added code snippets in welcome cloud app.

### CODE REVIEW

File: application.js [hunk 1]
```diff
@@ -16,6 +16,15 @@ app.use(cors());
 app.use('/sys', mbaasExpress.sys(securableEndpoints));
 app.use('/mbaas', mbaasExpress.mbaas);
 
+/* uncomment this code if you want to use $fh.auth in the app preview
+ * specify function for localAuth
```
Comment (wei-lee) [inline application.js]: perhaps explain this a little bit more clear. The localAuth is only used for local development. If the app is deployed on the platform, this function will be ignored and the request will be forwarded to the platform to perform authentication

File: application.js [hunk 2]
```diff
@@ -16,6 +16,20 @@ app.use(cors());
...
+ * localAuth is only used for local development. 
+ * If the app is deployed on the platform, 
+ * this function will be ignored and the request will be forwarded 
+ * to the platform to perform authentication.
+ * 
+
+app.use('/box', mbaasExpress.auth({localAuth: function(req, cb){
+  return cb(null, {status:401, body: {"message": "bad request"}});
+}}));
+
```
Comment (wei-lee) [inline application.js]: perhaps add an `or` here to make it clear that either way is ok.

**Comment by wtrocki** [comment]
@wei-lee @rachael-oregan Making this change to be part of the default cloud template is causing lots of confusion for the users. 

Can we remove this and ping documentation team to include it?  Providing such large commented snippet without documentation seems like anti pattern: https://github.com/ryanmcdermott/clean-code-javascript#comments 

Entire block is really misleading. Just uncommenting this piece of the code would not work and we missing documentation reference. 

FYI @PhilipGough @philbrookes @psturc - let's refer to documentation when testing auth.

**Comment by wei-lee** [comment]
@wtrocki Yeah, I agree this is not ideal. This commit is trying to resolve 2 issues:

1. $fh.auth is not working in the app preview. Because how $fh.auth works, if users want to try it in the app preview, they will need to add the uncomment code.
2. It doesn't work in local development mode. Again, it's the problem with how $fh.auth is implemented. Previously users will have to create the endpoint themselves to work around the problem. I was hoping this comment will make it easier for them.

If this is causing more problem than it resolves, maybe we can look at remove it. I think it's might be that we are missing more documentation about it? Perhaps create a more detailed doc page to explain it and add the link here?

**Comment by wtrocki** [comment]
I think it would be better is to document it and include link into the template, so we can have much more information where is that happening etc. Long comment here looks strange.

ping @sedroche

**Pull request merged**